src/bitcoind.cpp

```diff
@@ -254,6 +320,10 @@ static bool AppInit(NodeContext& node)
 
 MAIN_FUNCTION
 {
+#ifdef WAIT_FOR_DEBUGGER
+    if (int ret{HandleWaitForDebugger(argc, argv)}; ret != EXIT_SUCCESS) return ret;
```

Comment:
let's not propagate these further:
```suggestion
    if (int ret{HandleWaitForDebugger(std::span{argv + 1, size_t(argc - 1)})}; ret != EXIT_SUCCESS) return ret;
```